rust_snuba/src/processors/uptime_monitor_checks.rs

```diff
@@ -0,0 +1,157 @@
+use crate::config::ProcessorConfig;
+use anyhow::Context;
+use chrono::DateTime;
+use sentry_arroyo::backends::kafka::types::KafkaPayload;
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+use crate::types::{InsertBatch, KafkaMessageMetadata};
+
+pub fn process_message(
+    payload: KafkaPayload,
+    metadata: KafkaMessageMetadata,
+    _config: &ProcessorConfig,
+) -> anyhow::Result<InsertBatch> {
+    let payload_bytes = payload.payload().context("Expected payload")?;
+    let (rows, origin_timestamp) =
+        deserialize_message(payload_bytes, metadata.partition, metadata.offset)?;
+
+    InsertBatch::from_rows(rows, DateTime::from_timestamp(origin_timestamp as i64, 0))
+}
+
+pub fn deserialize_message(
+    payload: &[u8],
+    partition: u16,
+    offset: u64,
+) -> anyhow::Result<(Vec<UptimeMonitorCheckRow>, f64)> {
+    let monitor_message: UptimeMonitorCheckMessage = serde_json::from_slice(payload)?;
+
+    let rows = vec![UptimeMonitorCheckRow {
+        organization_id: 1,
+        project_id: 1,
+        environment: monitor_message.environment,
+        uptime_subscription_id: monitor_message.subscription_id,
+        uptime_check_id: monitor_message.guid,
+        scheduled_check_time: monitor_message.scheduled_check_time_ms,
+        timestamp: monitor_message.actual_check_time_ms,
+        duration: monitor_message.duration_ms,
+        region_slug: Some("global".to_string()),
+        check_status: monitor_message.status,
+        check_status_reason: monitor_message.status_reason.map(|r| r.r#type),
+        http_status_code: monitor_message
+            .request_info
+            .unwrap()
+            .http_status_code
+            .unwrap_or(0),
+        trace_id: monitor_message.trace_id,
+        retention_days: 30,
+        partition,
+        offset,
+    }];
+
+    Ok((rows, monitor_message.actual_check_time_ms))
+}
+
+#[derive(Debug, Deserialize)]
+struct UptimeMonitorCheckMessage {
+    // TODO: add these to the message
+    // organization_id: u64,
+    // project_id: u64,
+    // retention_days: u16,
+    // region_slug: Option<String>,
+    environment: Option<String>,
+    subscription_id: Uuid,
+    guid: Uuid,
+    scheduled_check_time_ms: f64,
+    actual_check_time_ms: f64,
+    duration_ms: u64,
+    status: String,
+    status_reason: Option<CheckStatusReason>,
+    trace_id: Uuid,
+    request_info: Option<RequestInfo>,
+}
+#[derive(Debug, Deserialize)]
+pub struct RequestInfo {
+    pub http_status_code: Option<u16>,
+}
+#[derive(Debug, PartialEq, Deserialize, Serialize)]
+#[serde(rename_all = "snake_case")]
+pub struct CheckStatusReason {
+    /// The type of the status reason
+    pub r#type: String,
```

Comment:
I'm sure it works but damn that's ugly. What happened to naming it `ty` and putting an `#[serde(rename = "type")]`...

Reply:
Also, it looks like it's a very static reason, we could benefit from having it be a `'a &str` instead of `String`.

Reply:
Not that big of a deal, saves on memory. We do it a lot in Relay too.

https://github.com/getsentry/relay/blob/master/relay-server/src/services/store.rs#L1255

Reply:
went ahead and converted all `String`s to `&str`s for all the strings! let me know if what i did makes sense, not super familiar with lifetimes in rust.